extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-preview.js

```diff
@@ -552,19 +555,45 @@ export class AmpStoryDevToolsTabPreview extends AMP.BaseElement {
     }).then(() => {
       deviceSpecs.player
         .getElement()
-        .addEventListener('storyNavigation', (event) => {
-          this.devices_.forEach((d) => {
-            if (d != deviceSpecs) {
-              d.player.show(null, event.detail.pageId);
-            }
-          });
-        });
+        .addEventListener('storyNavigation', (event) =>
+          this.onPlayerNavigation_(event, deviceSpecs)
+        );
       deviceSpecs.player.load();
     });
+    this.expectedNavigationEvents_[deviceSpecs.name] = [];
     this.devices_.push(deviceSpecs);
     this.updateDevicesInHash_();
   }
 
+  /**
+   * Triggered when a player emits a storyNavigationEvent.
+   *
+   * A navigation event from a player can come from a user interaction or a previous programmatic call.
+   * Expected navigation events from programmatic calls are stored in `this.expectedNavigationEvents_`,
+   * so they should not be propagated (but deleted from the list of expected events). If an event was not
+   * expected, it means it was user navigation and should be propagated to other players.
+   * @param {!Event} event
+   * @param {!DeviceInfo} deviceSpecs
+   * @private
+   */
+  onPlayerNavigation_(event, deviceSpecs) {
+    const {pageId} = event.detail;
+    const pageIndex = this.expectedNavigationEvents_[
```

Comment:
this `pageIndex` name is a bit confusing. We are storing these on the `expectedNavigationEvents` array. I think either we rename `pageIdx` to `eventPageIdx` or we rename `expectedNavigationEvents` to `expectedNavigationEventsForPageIdx`

Reply:
True, the second idea would be wrong because the pageIndex used here is just to know if the page is in the expectedNavigationEvents (by comparing it to -1) and then to clear all the expected events before it. `expectedNavigationEvents` is just a queue of navigations you expect the player to do (navigations from other players that haven't been reflected in this player yet), but fixing the issue of `.show` events not applying to players still loading. Can change the variable to `pageIndexInExpectedList`?